### PR TITLE
Increase unit test coverage for AWSMachine controller

### PR DESCRIPTION
**Pull request by Ankitasw**
**What type of PR is this?**
/area testing

**What this PR does / why we need it**:
This PR increases the unit tests coverage for AWSMachine controller. It also adds some of the CAPA specific integration tests. The integration tests are not in scope of this PR. Scenarios listed [here](https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/2910#issuecomment-1017527816).
Coverage of `controllers` pkg increased from 60% to 80%. 

**Which issue(s) this PR fixes (optional, in fixes #<issue number>(, fixes #<issue_number>, ...) format, will close the issue(s) when PR gets merged)**:
Part of #2910 

**Checklist**:

- [x] squashed commits
- [ ] includes documentation
- [X] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:
```release-note
Add unit tests for AWSMachine controller
```

### CODE REVIEW

File: controllers/awsmachine_controller.go [hunk 1]
```diff
@@ -525,7 +534,7 @@ func (r *AWSMachineReconciler) reconcileNormal(_ context.Context, machineScope *
 	}
 
 	// reconcile the deletion of the bootstrap data secret now that we have updated instance state
-	if deleteSecretErr := r.deleteEncryptedBootstrapDataSecret(machineScope, clusterScope); err != nil {
+	if deleteSecretErr := r.deleteEncryptedBootstrapDataSecret(machineScope, clusterScope); deleteSecretErr != nil {
```
Comment (sedefsavas) [inline controllers/awsmachine_controller.go:537]: good catch!

**Comment by sedefsavas** [comment]
Most tests in the integration file are unit tests. I will merge this so that you can refactor those under the open integration test PR.

Also, if we need a documentation for explaining the distinction between _unit-tests.go vs tests.go for developers (basically to explain why we have 2 separate files for tests), feel free to file an issue.

**Comment by Ankitasw** [comment]
> Most tests in the integration file are unit tests. I will merge this so that you can refactor those under the open integration test PR.
Also, if we need a documentation for explaining the distinction between _unit-tests.go vs tests.go for developers (basically to explain why we have 2 separate files for tests), feel free to file an issue.

I will file and issue for the same. Also, will refactor the tests in separate files.

**Pull request merged**